src/sc/ContractParam.js

```diff
@@ -62,13 +62,29 @@ class ContractParam {
    * Creates a ByteArray ContractParam.
    * @param {any} value
    * @param {string} format - The format that this value represents. Different formats are parsed differently.
+   * @param {any} args - Additional arguments such as decimal precision
    * @return {ContractParam}
    */
-  static byteArray (value, format) {
+  static byteArray (value, format, ...args) {
     if (format) format = format.toLowerCase()
     if (format === 'address') {
       return new ContractParam('ByteArray', reverseHex(getScriptHashFromAddress(value)))
     } else if (format === 'fixed8') {
+      var decimals = 8
+      if (args.length === 1) {
+        decimals = args[0]
+      }
+      if (!isFinite(value)) throw new Error(`Input should be number!`)
+      var e = 1
+      var p = 0
+      while (Math.round(value * e) / e !== value) {
+        e *= 10
+        p++
+      }
+      if (p > decimals) throw new Error(`Input value exceeds maximum precision!`)
+      value = new Fixed8(value)
+      let divisor = new Fixed8(Math.pow(10, 8 - decimals))
+      value /= divisor

```

Comment:
You cannot use normal arithmetic functions with Fixed8. You will have to use `value = value.div(divisor)`.